### ISSUE TITLE
Unable to put proposal on chain?

### ISSUE THREAD

**Issue by robliou** [AxLabs/grantshares#55]
Hi,

I'm attempting to put my proposal (dApp- Gamefied Forecasting + Advanced Analytics Dashboard) on chain, but when I click the "Put On-Chain" button, nothing happens. I'm using the Firefox browser. The other buttons on the page work fine.

Looking at the dev tools under the console tab, this is the error that occurs every time I click the button:

"TypeError: Expected String  NextJS   f decode  getScrip... "

Using other browsers (i.e. Chrome, Edge), I can access the Grantshares homepage, but none of the proposals render. Therefore, I've been using Firefox this whole time.

Thank you,

**Comment by gsmachado** [comment]
Hello @robliou 

Thanks for taking the time to create an issue!

Yes, I can reproduce the issue with Firefox [version 108.0.2 (64-bit)]. We'll investigate.

**Comment by gsmachado** [comment]
@robliou In addition: you mentioned that using other browsers (like Edge and Chrome) you can't use the dApp, at all. I assume you mean that https://grantshares.io works, but not https://grantshares.io/app/proposals.

So, a couple of questions to better understand what's happening:

1) Do you have any kind of ad blocker running on Chrome or Edge?
2) Can you please screenshot the console error and provide it to us (here in the issue)?
3) Do you have NeoLine installed in your Chrome?
4) What is the Edge version?
5) What is the Chrome version?

**Comment by gsmachado** [comment]
@csmuller and @alexsteixeira: FYI, read this issue.

**Comment by gsmachado** [comment]
@robliou BTW: I can't reproduce the behavior you described with Edge and Chrome. I can normally navigate on the [GrantShares dApp](https://grantshares.io/app/proposals) using Edge (on Windows) and Chrome (on Windows and MacOS).

**Comment by gsmachado** [comment]
> Hello @robliou
> 
> Thanks for taking the time to create an issue!
> 
> Yes, I can reproduce the issue with Firefox [version 108.0.2 (64-bit)]. We'll investigate.

It seems that this problem ☝️ is related to Windows.

I can reproduce the error when trying to "Put Proposal on-chain" with Chrome, Firefox, and Edge, on Windows 11. All browsers are at the latest and greatest version.

![image](https://user-images.githubusercontent.com/2124276/211162695-b7c25855-96f6-4e2a-81d2-71e52c5f20a5.png)

**Comment by robliou** [comment]
Hi @gsmachado ,

Please see my responses below: 

> 
>     1. Do you have any kind of ad blocker running on Chrome or Edge?
       
         For Chrome, yes. I turned it off though, still got the same error (see answer for #2).

         For Edge, no ad blocker.


> 
>     2. Can you please screenshot the console error and provide it to us (here in the issue)?
     
       For Chrome, please see the screenshot below:


![A](https://user-images.githubusercontent.com/81087205/211194803-daf942d9-84b1-4c91-b2fa-d65901e398da.png)
> 
>     3. Do you have NeoLine installed in your Chrome?

        Yes, I have NeoLine in Chrome. I can connect to it just fine, but the proposals still don't show. I use Github to connect in Firefox.
> 
>     4. What is the Edge version?

Edge: Version 108.0.1462.76 (Official build) (64-bit)
> 
>     5. What is the Chrome version?

Chrome: Version 108.0.5359.125 (Official Build) (64-bit)

**Comment by robliou** [comment]
> 

Wow, great. Yes, the error you are showing here is similar to the one I am getting in Firefox:

![A](https://user-images.githubusercontent.com/81087205/211195071-cd597389-68ba-42cb-8f98-ed5bbca78f47.png)

Appreciate your quick response over the weekend on this! @gsmachado  What can be done?

Thanks,

**Comment by gsmachado** [comment]
> Appreciate your quick response over the weekend on this! @gsmachado What can be done?

@robliou it's the first case where someone reports that Chome & Firefox & Edge are not working for actions in the dApp. This is not usual. I suspect that the problem in Chrome and Edge is something specific to your setup... but the one in Firefox is something I can also reproduce, so, a general problem.

Unfortunately, nothing you can do at the moment. Give us a bit of time to come back to you.

**Comment by csmuller** [comment]
@robliou, Before you click the "Put Proposal on-chain" button you need to connect with a wallet. We neglected showing "connect wallet" prompt automatically if you are not connected and try clicking that button (we'll work on that).  

I assume that, once your wallet is connected, it will work fine with Firefox on your machine. As far as i understood, you don't have the CORS issue in Firefox that you experience in Chrome and Edge.
For the CORS issues we have to look into it in more detail. I don't know yet what the problem could be. The error message doesn't seem to reflect how the system is actually configured.

**Comment by gsmachado** [comment]
@robliou can you please:

1) Open a Chrome browser and go to https://grantshares.io
2) Enable "Developer Tools"
3) In the "Developer Tools" pane, go to "Network tab". Make sure you select "Fetch/XHR" to filter out some stuff.
4) Click on "Launch dApp" on the GrantShares landing page
5) Check the "Network tab" and search for the network request starting with `all?page=0`.
6) Take a snapshot with the "Request" and "Response" visible. Like the one below:

![Screenshot 2023-01-09 at 14 47 46](https://user-images.githubusercontent.com/2124276/211322917-da2e46a4-2470-4cb8-9c42-22bd17aa751a.png)

7) Post the snapshot to us here in the issue! 🙏 

Thanks in advance. With this info we will be able to debug the problem a bit better. 👍

**Comment by robliou** [comment]
@csmuller, thanks for the tip about needing to connect my wallet first! I was using Github to connect my profile in Firefox, and no luck.

As you're well aware, Firefox doesn't support wallets, so putting a _proposal on-chain currently requires Chrome only_?

To get around this issue for now, I used my wife's old Macbook and installed my Neoline wallet in her Chrome Browser; then logged in and pushed the proposal on-chain. Thanks for your help!

**Comment by gsmachado** [comment]
> To get around this issue for now, I used my wife's old Macbook and installed my Neoline wallet in her Chrome Browser; then logged in and pushed the proposal on-chain. Thanks for your help!

😮 omg, I hope you saved the private key. 😅 

Be careful, and congrats on pushing the proposal on-chain!

**Comment by robliou** [comment]
> @robliou can you please:
> 
>     1. Open a Chrome browser and go to https://grantshares.io
> 
>     2. Enable "Developer Tools"
> 
>     3. In the "Developer Tools" pane, go to "Network tab". Make sure you select "Fetch/XHR" to filter out some stuff.
> 
>     4. Click on "Launch dApp" on the GrantShares landing page
> 
>     5. Check the "Network tab" and search for the network request starting with `all?page=0`.
> 
>     6. Take a snapshot with the "Request" and "Response" visible. Like the one below:
> 
> 
> ![Screenshot 2023-01-09 at 14 47 46](https://user-images.githubusercontent.com/2124276/211322917-da2e46a4-2470-4cb8-9c42-22bd17aa751a.png)
> 
>     7. Post the snapshot to us here in the issue! 🙏
> 
> 
> Thanks in advance. With this info we will be able to debug the problem a bit better. 👍

@gsmachado No problem! Please see below:

![A](https://user-images.githubusercontent.com/81087205/211331219-17ceba8a-4735-4aa0-ae98-ef1940bbbe11.png)

![A2](https://user-images.githubusercontent.com/81087205/211331263-09bd0ce0-cb25-47f4-b9f2-1a0379698097.png)

**Comment by robliou** [comment]
> > To get around this issue for now, I used my wife's old Macbook and installed my Neoline wallet in her Chrome Browser; then logged in and pushed the proposal on-chain. Thanks for your help!
> 
> 😮 omg, I hope you saved the private key. 😅
> 
> Be careful, and congrats on pushing the proposal on-chain!

Yes, thank you sir! I have a pretty good relationship with my wife thus far (we sleep in the same bed), so I'm not too worried if she runs off with the private key, lol. 

Good point though, I'll delete that extra wallet once this is all completed!

**Comment by gsmachado** [comment]
I'm closing this issue since it's old and nobody has recently complained about this.

However, I'd say that we should include Firefox, Safari, and Edge browsers in our "smoke test" suite.

@OT-kraftchain I created this issue, which we should follow up and accomplish: https://github.com/AxLabs/grantshares-app/issues/625
